src/llms/llm_loader.py

```diff
@@ -0,0 +1,233 @@
+# workaround to disable UserWarning
+import warnings
+warnings.simplefilter("ignore", UserWarning)
+
+from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler
+from langchain.callbacks.manager import CallbackManager
+
+import os, inspect
+
+from utils.logger import Logger
+
+class LLMLoader:
+    """
+    Note: This class loads the LLM backend libraries if the specific LLM is loaded.
+    Known caveats: Currently supports a single instance/model per backend
+
+    The llm_backends parameter is an array with the backend llm and (optional) parameters to override
+
+    # format for dictionary
+    llm_backend = {
+        'backend': 'openai',
+        'params': {...}     # optional key
+    }
+
+    # overriding specific parameters
+    llm_backend = {
+        'backend' : 'ollama',
+        'params': {'temperature': 0.02, 'top_p': 0.95},
+    }
+
+    # using default parameters
+    llm_backend = {'backend': 'tgi'}
+
+    # Using the class
+    llm_config = LLMLoader(llm_backend=llm_backend)
+    llm_chain = LLMChain(llm=llm_config.llm, prompt=prompt)
```

Comment:
maybe this is a python idioms thing, but this flow still looks weird to me.

Why have LLMLoader return an llm_config and not just return an llm?  (don't make LLMLoader a class, it can just be a function that instantiates the desired llm implementation, using the passed in or loaded configuration/parameters/arguments)

The only reason i can see to return a wrapper object is if you can use that wrapper object to instantiate more llms (either of the same provider implementation but different args, or a different provider impl entirely).

as it is you seem to be returning an LLMLoader wrapper class just so someone can use the `.llm` instance it contains.  The wrapper doesn't seem to serve a purpose (maybe you have a future purpose in mind?)

Reply:
We can return the LLM class or just do `LLMLoader(llm_backend=...).llm` but you won't be able to introspect anything about that instance of the class. For example, lets say that we want to add the capability of modifying parameters or reload the LLM object or to have a proper status function that validates the LLM backend is still operational. It all depends on preferences and of any additional methods we add to the class.

Reply:
Fair enough.  I struggle a bit w/ calling it "LLMLoader" when it sounds like your future usage is more like "LLMWrapper" (wraps the native langchain llm instance w/ something that abstracts the implementation and provides a common interface for checking status or modifying config), but that's a bit of a bikeshed.